src/nunit-gui/Model/ResultSummaryReporter.cs

```diff
@@ -0,0 +1,87 @@
+// ***********************************************************************
+// Copyright (c) 2017 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System.Globalization;
+using System.IO;
+using System.Text;
+
+namespace NUnit.Gui.Model
+{
+    public static class ResultSummaryReporter
+    {
+        public static string WriteSummaryReport(ResultSummary summary)
+        {
+            var sb = new StringBuilder();
+            using (var writer = new StringWriter(sb))
```

Comment:
To be honest I was wondering about the same thing my self. I decided in the end to keep the string writer and color around because of how it is done in the console. I've seen a few examples of where rich text was allowed in the tooltip and if we decide to make a WPF implementation then it would be nice to copy...

then again it violates YAGNI...*sigh* ok, i'll simplify.